### PR TITLE
Check optional parameter

### PR DESCRIPTION
**Pull request by joswarmer**
Crash with infinite depth retrieve fixed.

### CODE REVIEW

File: packages/test/src/test/RepositoryClient.ts [hunk 1]
```diff
@@ -62,10 +62,15 @@ export class RepositoryClient {
 
     async testRetrieve(nodeIds: string[], depth?: number) {
         console.log(`test.testRetrieve ${nodeIds} with depth ${depth}`)
-        depth = depth || 999
+        let params: string
```
Comment (dslmeinte) [inline packages/test/src/test/RepositoryClient.ts:65]: const params = (depth === undefined) ? "" : `depthLimit=${depth}`?

File: packages/core/src/controllers/LionWebBulkApi.ts [hunk 2]
```diff
@@ -54,7 +54,13 @@ class LionWebBulkApiImpl implements LionWebBulkApi {
     async retrieve(req: Request, res: Response): Promise<void> {
         console.log("LionWebBulkApiImpl.retrieve: ")
         const mode = req.query["mode"] as string
-        const depthLimit = Number.parseInt(req.query["depthLimit"] as string)
+        const depthParam = req.query["depthLimit"];
+        let depthLimit: number
```
Comment (dslmeinte) [inline packages/core/src/controllers/LionWebBulkApi.ts:58]: `const depthLimit = (typeof depthParam === "string) ? Number.parseInt(depthParam) : Number.MAX_SAFE_INTEGER`? And: `Number.parseInt` returning `NaN` is not handled now.

**Review by dslmeinte — CHANGES_REQUESTED** [review]

File: packages/test/src/test/RepositoryClient.ts (hunk 1 above)
Comment (joswarmer) [reply packages/test/src/test/RepositoryClient.ts:65]: Much more elegant indeed.

File: packages/core/src/controllers/LionWebBulkApi.ts (hunk 2 above)
Comment (joswarmer) [reply packages/core/src/controllers/LionWebBulkApi.ts:58]: NaN is handled at line 66.

Comment (joswarmer) [reply packages/core/src/controllers/LionWebBulkApi.ts:58]: Updated code as described.

**Review by dslmeinte — APPROVED** [review]

**Pull request merged**